### ISSUE TITLE
Automation for citing Coq-community projects in publications

### ISSUE THREAD

**Issue by palmskog** [coq-community/manifesto#75]
## Meta-issue ##

As highlighted for Coq in [this issue](https://github.com/coq/coq/issues/5766), the possibility of conveniently citing projects in academic publications is important for many community participants, and can lead to higher visibility and more funding in the future.

The typical way of making an open source software project citeable is to put a release on Zenodo, e.g., [Coq 8.9.0](https://zenodo.org/record/2554024) and [Equations 1.0](https://zenodo.org/record/1117297). This generates a DOI and convenient citation metadata (e.g., BiBTeX).

If we could automate putting Coq-community releases on Zenodo, I think this would be a good idea. It would also mean one more preservation archive besides GitHub.

**Comment by Zimmi48** [comment]
That's a very good idea. It shouldn't be much more work to implement this automation for all coq-community packages beyond just for Coq, and it should even be useful for many more projects.

**Comment by Zimmi48** [comment]
Actually, I had almost forgotten that Zenodo already provides this kind of automation (as mentioned in the Coq issue): https://zenodo.org/account/settings/github/
I didn't find it sufficiently satisfying for Coq, but it should be for most projects, including for coq-community projects. We could start testing it on a sample of projects, and then document how to use it.

**Comment by palmskog** [comment]
GitHub has now added a way for flagging up citation options in the sidebar for a project. This sidebar can for example provide BiBTeX boilerplate. See more here: https://twitter.com/natfriedman/status/1420122675813441540

The gist is that one needs to push a file called `CITATION.cff` containing the desired metadata. This file could for example be automatically pushed for a coq-community repo once a new release has been done and submitted to Zenodo.

**Comment by aubertc** [comment]
Any progress on this? Having `CITATION.cff` for each package would be very nice, if you provide some kind of guidance on what information should be included, I'm happy to open a couple of pull requests for some packages.

The [documentation](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files) for github's integration and the [schema spefication](https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md) are fairly clear, and I suppose each package manager can tweak the file to their liking, but maybe some organization-level guidance would be nice.

For example, [`contact`](https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md#contact) could be systematically used for the package manager.

**Comment by aubertc** [comment]
Example at https://github.com/rocq-community/coqtail-math/pull/10

I don't know how much of that can be automated to be honest.

*Note that I'm not discussing the "another preservation archive besides GitHub" bit of the original issue*

**Comment by Zimmi48** [comment]
Thanks for starting to work on this! Let's forget about Zenodo indeed as a prerequisite for generating a `CITATION.cff` file. To me, it looks like we could easily extend the [Rocq-community templates](https://github.com/rocq-community/templates) to generate these files automatically. The only annoyance is filling the part with a reference to the latest version, and keeping the file up-to-date when new versions are released. We can also have support for the `preferred-citation` field in case a paper is available. We already have a field for associated DOIs that would need to be extended a little bit.

**Comment by aubertc** [comment]
Yes, this seems like a good way to go. Happy to take a stab at it, only problems are:
1. What about existing projects? Do they re-generate their files using the templates periodically, or will they still have to manually add this `CITATION.cff` file?
2. This "version book keeping" is annoying indeed. Best *may* be to simply leave it blank, and have maintainers update it / integrate it in their workflow if they so wish. If there was some automation for the releases built-in the template, then maybe it could be added, but since this part is not integrated, I don't see an easy way out.

**Comment by Zimmi48** [comment]
BTW, related issue here: https://github.com/rocq-community/templates/issues/129

> Do they re-generate their files using the templates periodically

Yes, they usually do, but they are not forced to use the templates for every file (it's also supported to use the templates on a per-file basis).


> This "version book keeping" is annoying indeed. Best _may_ be to simply leave it blank, and have maintainers update it / integrate it in their workflow if they so wish. If there was some automation for the releases built-in the template, then maybe it could be added, but since this part is not integrated, I don't see an easy way out.

Release automation would indeed be nice to have, but we don't have anything like it yet. Several possible directions to make it happen would be to use a GitHub Actions-based workflow for release automation, or to use the [Rocq bot](https://github.com/rocq-prover/bot/). I thought we had an issue discussing release automation in Rocq-community somewhere, but I cannot find it anymore.
